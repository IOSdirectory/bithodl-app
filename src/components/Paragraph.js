import React, { Component } from 'react';
import { StyleSheet } from 'react-native';
import PropTypes from 'prop-types';
import StyledText from './StyledText';

const styles = StyleSheet.create({
  paragraph: {
    color: '#8F8E94',
    fontSize: 16,
    marginBottom: 20,
    textAlign: 'center'
  }
});

export default class Paragraph extends Component {
  render() {
    return (
      <StyledText style={[styles.paragraph, this.props.style]}>
        {this.props.children}
      </StyledText>
    );
  }
}

Paragraph.propTypes = {
  style: PropTypes.any,
  children: PropTypes.node
};
