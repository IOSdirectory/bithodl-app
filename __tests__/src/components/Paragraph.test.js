import React from 'react';
import { Text } from 'react-native';
import renderer from 'react-test-renderer';
import Paragraph from '../../../src/components/Paragraph';

describe('Paragraph', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <Paragraph>
        <Text>42193c11-0e46-44c7-9406-3a74aea01145</Text>
      </Paragraph>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
