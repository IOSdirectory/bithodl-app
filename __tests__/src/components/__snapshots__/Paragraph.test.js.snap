// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  numberOfLines={undefined}
  style={
    Array [
      Object {
        "color": "#504F63",
        "fontFamily": "Avenir Next",
        "fontWeight": "400",
      },
      Array [
        Object {
          "color": "#8F8E94",
          "fontSize": 16,
          "marginBottom": 20,
          "textAlign": "center",
        },
        undefined,
      ],
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
  >
    42193c11-0e46-44c7-9406-3a74aea01145
  </Text>
</Text>
`;
